File: CPP/mixer.cpp
#include <iostream>
using namespace std ;

#include "mixer.h"

void Mixer::render() {
	cout << "M" ;
}

int Mixer::getIdentity() {
	return 4 ;
}

void Mixer::Interact(int i){
	cout << "Masukkan command 'Mix' untuk menggunakan Mixer" << endl << endl;
}

void Mixer::Mix(){
	showListOfSideProduct();
	
}

void Mixer::showListOfSideProduct(){
	cout << "11. Beef Rolade" <<endl; //CowMeat + ChickenEgg
	cout << "12. Butter" << endl; // 2 Milk
	cout << "13. Omelet" << endl; // ChickenEgg + Milk
}

File: GUI-nya/JavaGUI/src/Mixer.java
import java.util.*;

class Mixer extends Facility{
	private List<String> resepList;
	
	public char render() {
		return 'M' ; //System.out.print( "M" );
	}

	public int getIdentity() {
		return 4 ;
	}

	public void interact(int i){
		System.out.println( "Masukkan command 'Mix' untuk menggunakan Mixer" );
	}

	public void mix(){
		showListOfSideProduct();
	}

	public void showListOfSideProduct(){
		for(String str : resepList){
			System.out.println( str ); 
		}
	}
	
	public Mixer(){
		ArrayList<String> resepList = new ArrayList<String>();
		resepList.add("11. Beef Rolade"); //CowMeat + ChickenEgg
		resepList.add("12. Butter"); // ChickenEgg + CowMilk
		resepList.add("13. Omelet"); // DuckEgg + CowMilk
	}
}
